plugin/core/configurations.py

```diff
@@ -44,13 +44,18 @@ def match_view(self, view: sublime.View, include_disabled: bool = False) -> Gene
         """
         Yields configurations matching with the language's document_selector
         """
-        configs = self.match_document(view2scope(view))
-        if include_disabled:
-            yield from configs
-        else:
-            for config in configs:
-                if config.enabled:
-                    yield config
+        try:
+            configs = self.match_document(view2scope(view))
+            if include_disabled:
+                yield from configs
+            else:
+                for config in configs:
+                    if config.enabled:
+                        yield config
+        except IndexError:
+            # We're in the worker thread, and the view is already closed. This means view.scope_name(0) returns an
+            # empty string.
```

Comment:
General note: it's not always obvious whether a method runs on the main or separate thread. Maybe we could annotate methods with some custom decorator that would assert main/non-main thread? It would make things clearer for us (me) and do a runtime check in case it's used in the way it's not supposed to.

Reply:
There are three types of threads for us to worry about: the main thread, the ST worker thread, and our own spawned threads. I've started adding an `_async` suffix to functions that should run on the ST worker thread, but I don't have any runtime checks. It is perhaps a possibility to add a decorator for this (in another PR).